.jenkins/pytorch/common.sh

```diff
@@ -181,11 +181,12 @@ function get_exit_code() {
 }
 
 function file_diff_from_base() {
+  BASE_REF=${BASE_REF:-master}
   # The fetch may fail on Docker hosts, but it's not always necessary.
   set +e
-  git fetch origin master --quiet
+  git fetch origin "${BASE_REF}" --quiet
   set -e
-  git diff --name-only "$(git merge-base origin master HEAD)" > "$1"
```

Comment:
what was this doing before?

Like...a three way merge between origin (a remote?) master and HEAD?

Reply:
The inline `$(git merge-base origin master HEAD)` is finding the most recent commit between `master` and our current `HEAD`, which will output some SHA like `abc1234`

Then the `git diff --name-only abc1234` will output all of the files names changed for that particular commit.

After writing this out I can see that this is flawed and what we really should be doing is just:

`git diff --name-only ${BASE_REF}..HEAD`

Reply:
If I try this in my local repo: `git merge-base origin master HEAD`
I get:
`fatal: Not a valid object name origin`

Reply:
Just changed it to do a simple `git diff` with a range instead of comparing against the merge base

Reply:
put another way: https://github.com/pytorch/pytorch/pull/36262 looks like it works but https://github.com/pytorch/pytorch/pull/36255 doesn't.

Reply:
> If I try this in my local repo: `git merge-base origin master HEAD`
> I get:
> `fatal: Not a valid object name origin`

This could also be because you don't have a reference `master` for your `origin` remote?

I get the exact same error when running `git merge-base upstream master HEAD` since there's no reference `upstream` that's tracking my `upstream` remote

Reply:
I think it's actually that the ci environment has an origin branch for some reason.